### DISCUSSION TITLE
temp. texture FileNotFoundError when running pyinstaller exe

### DISCUSSION THREAD

**Discussion by jj-foster** [marcomusy/vedo#615] [Q&A]
Hi, 

I need to 'compile' a script containing vedo.Mesh to an exe. I need it to be an exe so it is accessible for other people who do not have python and all the script requirements installed.

When I run the exe I get a FileNotFoundError:
>Traceback (most recent call last):
  File "mesh_generator.py", line 15, in <module>
    from vedo import Mesh
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 495, in exec_module
  File "vedo\__init__.py", line 120, in <module>
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\USER\\AppData\\Local\\Temp\\_MEI38122\\vedo\\textures/'
[13192] Failed to execute script 'mesh_generator' due to unhandled exception!

I'm using pyinstaller btw.

I don't really understand what's going on here. Any help would be appreciated :)

Thanks

**Comment by marcomusy** [comment]
I never tried pyinstaller.. in any case the error seems related to the fact that a temporary directory is created which the system cannot access, something related to textures...

**Reply by marcomusy** [reply]
hi again, try
`pip install -U git+https://github.com/marcomusy/vedo.git`
then have a look at
https://github.com/marcomusy/welsh_embryo_stager/blob/main/stager.spec

**Comment by Michelegre** [comment]
Did you fix it? Because I have the same problem and I don't know how to solve it

**Reply by marcomusy** [reply]
Refer to #689

**Comment by RubendeBruin** [comment]
the data of vedo needs to be distributed with the package.

You can copy it manually after creating the executable or add it to the "datas" field in the .spec file:

```
a = Analysis(['DAVEgui.py'],
             pathex=['C:\\data\\Dave\\standalone'],
             binaries=[],
	     datas=[(r'C:\python\miniconda3\envs\builddave\Lib\site-packages\vedo\fonts\*','./vedo/fonts')],
             hiddenimports=hidden,
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
```

also make sure that you have installed `pyinstaller-hooks-contrib` or else you'll run into import errors on the vtk modules.
